File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/validator/ValidationMetricsCollector.java
package cz.fb.manaus.reactor.betting.validator;

import com.codahale.metrics.MetricRegistry;
import com.google.common.base.Joiner;
import cz.fb.manaus.core.model.Side;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ValidationMetricsCollector {

    public static final String PREFIX = "validator.stats";
    @Autowired
    private MetricRegistry metricRegistry;

    public void updateMetrics(ValidationResult result, Side type, String validatorName) {
        var name = getName(type, result.isSuccess(), validatorName);
        metricRegistry.counter(name).inc();
    }

    private String getName(Side type, boolean pass, String validatorName) {
        return Joiner.on('.').join(PREFIX,
                type.name().toLowerCase(), validatorName,
                pass ? "pass" : "fail");
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/betting/validator/ValidationMetricsCollector.kt
package cz.fb.manaus.reactor.betting.validator

import cz.fb.manaus.core.model.Side
import io.micrometer.core.instrument.Metrics
import org.springframework.stereotype.Component

@Component
object ValidationMetricsCollector {

    fun updateMetrics(result: ValidationResult, type: Side, validatorName: String) {
        Metrics.counter(
            "mns_validator_stats",
            "side", type.name.lowercase(),
            "validator", validatorName,
            "result", result.name.lowercase()
        ).increment()
    }
}
